tools/harness_capability_profiles.py

```
@@ -609,6 +654,98 @@ def markdown_table_rows(markdown: str, heading: str) -> list[dict[str, str]]:
     return rows
 
 
+def validate_required_markdown_sections(markdown: str, required_sections: list[str], prefix: str, path: Path) -> list[CheckResult]:
+    results: list[CheckResult] = []
+    for section in required_sections:
+        if section not in markdown:
+            results.append(CheckResult(f"{prefix}:section:{section.removeprefix('## ').casefold().replace(' ', '_')}", False, f"missing section: {section}", path.as_posix()))
```

Ruff (0.15.12)

[warning] Use a list comprehension to create a transformed list

(PERF401)